### PR TITLE
PSR-1531 fixed change mode for member details page

### PR DESCRIPTION
**Pull request by narendrapaduchuri**

### CODE REVIEW

File: app/controllers/nonsipp/memberdetails/SchemeMemberDetailsAnswersController.scala [hunk 1]
```diff
@@ -110,23 +110,26 @@ class SchemeMemberDetailsAnswersController @Inject()(
...
+      //      val memberDetailsChanged: Boolean = (for {
+      //        initial <- request.pureUserAnswers
+      //        initialMemberPayments <- initial.buildMemberDetails(srn, index)
+      //        currentMemberPayments <- request.userAnswers.buildMemberDetails(srn, index)
+      //      } yield initialMemberPayments != currentMemberPayments).getOrElse(false)
+
+      lazy val memberDetailsChanged =
+        request.userAnswers.changed(_.buildMemberDetails(srn, index))
 
       lazy val justAdded = mode.isNormalMode && request.userAnswers.get(MemberStatus(srn, index)).isEmpty
```
Comment (kberec) [inline app/controllers/nonsipp/memberdetails/SchemeMemberDetailsAnswersController.scala:122]: Is it maybe that the justAdded flag should be true in NormalMode and CheckMode too?

File: test/controllers/nonsipp/memberdetails/SchemeMemberDetailsAnswersControllerSpec.scala [hunk 2]
```diff
@@ -223,10 +223,10 @@ class SchemeMemberDetailsAnswersControllerSpec extends ControllerBaseSpec {
             pureUserAnswers = userAnswersWithNino.unsafeSet(MemberStatus(srn, refineMV(1)), MemberState.New).some,
             expectations = (ua: UserAnswers) =>
               List(
-                ua.get(MemberStatus(srn, refineMV(1))).exists(_.changed),
+                ua.get(MemberStatus(srn, refineMV(1))).exists(_._new),
                 ua.get(SafeToHardDelete(srn, refineMV(1))).isEmpty
               )
-          ).withName(s"Set Member status to Changed when a member detail has changed")
+          ).withName(s"Set Member status to New when a member detail has changed")
```
Comment (westwater) [inline test/controllers/nonsipp/memberdetails/SchemeMemberDetailsAnswersControllerSpec.scala:229]: the name of this test doesn't make sense, also this test should have stayed the same, if it failed it means something is missing (completed page is missing from test useranswers / wrong mode, etc)

If anything a new test case should have been written for the additional logic

Comment (narendrapaduchuri) [reply test/controllers/nonsipp/memberdetails/SchemeMemberDetailsAnswersControllerSpec.scala]: Hi @westwater, could you please check now, thanks

Comment (kberec) [reply test/controllers/nonsipp/memberdetails/SchemeMemberDetailsAnswersControllerSpec.scala]: @narendrapaduchuri I think the second part of @westwater's comment is not addressed: 
"If anything a new test case should have been written for the additional logic", in my understanding that is to add a test where `call = onSubmit(CheckMode)` and MemberDetailsCompletedPage is empty in user answers. To check the additional logic you have added.

**Review by kberec — APPROVED** [review]
